packages/core/src/parseAmplifyGen2.ts

```diff
@@ -0,0 +1,245 @@
+import { OAuthProvider } from './libraryUtils';
+import { Gen2AuthProperties } from './singleton/gen2/types';
+import { Gen2Config, LegacyConfig, ResourcesConfig } from './singleton/types';
+
+export function isGen2Config(
+	resourceConfig: ResourcesConfig | LegacyConfig | Gen2Config,
```

Comment:
Similar comment to below but it didn't strike me as a big deal until the function body grew and the parameter was oft used. It does make even more sense here since the incoming parameter can be any type of config.
```suggestion
	config: ResourcesConfig | LegacyConfig | Gen2Config,
```